common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt

```diff
@@ -0,0 +1,227 @@
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.ViewState
+import com.duckduckgo.mobile.android.ui.view.gone
+import com.duckduckgo.mobile.android.ui.view.show
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class NotifyMeView : FrameLayout {
+
+    constructor(context: Context) : this(context, null)
+
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+    ) : this(
+        context,
+        attrs,
+        R.style.Widget_DuckDuckGo_NotifyMeView,
+    )
+
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyle: Int,
+    ) : super(context, attrs, defStyle) {
+
+        val attributes = context.obtainStyledAttributes(attrs, R.styleable.NotifyMeView)
+
+        setTitle(attributes.getString(R.styleable.NotifyMeView_titleText) ?: "")
+        setSubtitle(attributes.getString(R.styleable.NotifyMeView_subtitleText) ?: "")
+
+        binding.notifyMeClose.setOnClickListener {
+            viewModel.onCloseButtonClicked()
+        }
+
+        binding.notifyMeButton.setOnClickListener {
+            viewModel.onNotifyMeButtonClicked()
+        }
+
+        attributes.recycle()
+    }
+
+    private var listener: NotifyMeListener? = null
+    private var coroutineScope: CoroutineScope? = null
+
+    private val binding: ViewNotifyMeViewBinding by viewBinding()
+
+    private val viewModel by lazy {
+        val factory = NotifyMeViewModel.Factory(findViewTreeSavedStateRegistryOwner()!!)
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, factory)[NotifyMeViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.addObserver(viewModel)
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+
+        viewModel.setNotifyMeListener(listener)
+
+        viewModel.viewState
+            .onEach { render(it) }
+            .launchIn(coroutineScope!!)
+
+        viewModel.commands()
+            .onEach { processCommands(it) }
+            .launchIn(coroutineScope!!)
+
+        checkNotificationsPermissions()
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.removeObserver(viewModel)
+
+        viewModel.removeNotifyMeListener()
+
+        listener = null
+
+        coroutineScope?.cancel()
+        coroutineScope = null
+    }
+
+    fun setListener(listener: NotifyMeListener?) {
+        this.listener = listener
+    }
+
+    fun setTitle(title: String) {
```

Comment:
If you change the attribute name as suggested, remember to change the API to `setPrimaryText` / `setSecondaryText`

Reply:
Updated.